include/vast/Translation/CodeGenUnsupportedVisitor.hpp

```diff
@@ -0,0 +1,527 @@
+		VISITOR_STMT(GNUNullExpr)
+
+		VISITOR_STMT(GotoStmt)
+		VISITOR_STMT(IfStmt)
+		VISITOR_STMT(ImaginaryLiteral)
+		VISITOR_STMT(ImplicitCastExpr)
+		VISITOR_STMT(IndirectGotoStmt)
+		VISITOR_STMT(InitListExpr)
+		VISITOR_STMT(IntegerLiteral)
+		VISITOR_STMT(LabelStmt)
+		VISITOR_STMT(LambdaExpr)
+		VISITOR_STMT(MSAsmStmt)
+		VISITOR_STMT(MSDependentExistsStmt)
+		VISITOR_STMT(MSPropertyRefExpr)
+		VISITOR_STMT(MSPropertySubscriptExpr)
+		VISITOR_STMT(MaterializeTemporaryExpr)
+		VISITOR_STMT(MatrixSubscriptExpr)
+		VISITOR_STMT(MemberExpr)
+		VISITOR_STMT(NoInitExpr)
+		VISITOR_STMT(NullStmt)
+		VISITOR_STMT(OffsetOfExpr)
+		VISITOR_STMT(OpaqueValueExpr)
+		VISITOR_STMT(PackExpansionExpr)
+		VISITOR_STMT(ParenExpr)
+		VISITOR_STMT(ParenListExpr)
+
+		VISITOR_STMT(PseudoObjectExpr)
+		VISITOR_STMT(RecoveryExpr)
+		VISITOR_STMT(RequiresExpr)
+		VISITOR_STMT(ReturnStmt)
+		VISITOR_STMT(SEHExceptStmt)
+		VISITOR_STMT(SEHFinallyStmt)
+		VISITOR_STMT(SEHLeaveStmt)
+		VISITOR_STMT(SEHTryStmt)
+		VISITOR_STMT(SYCLUniqueStableNameExpr)
+		VISITOR_STMT(ShuffleVectorExpr)
+		VISITOR_STMT(SizeOfPackExpr)
+		VISITOR_STMT(SourceLocExpr)
+		VISITOR_STMT(StmtExpr)
+		VISITOR_STMT(StringLiteral)
+		VISITOR_STMT(SubstNonTypeTemplateParmExpr)
+		VISITOR_STMT(SubstNonTypeTemplateParmPackExpr)
+		VISITOR_STMT(SwitchStmt)
+		VISITOR_STMT(TypeTraitExpr)
+		VISITOR_STMT(TypoExpr)
+		VISITOR_STMT(UnaryExprOrTypeTraitExpr)
+
+		VISITOR_STMT(UnresolvedLookupExpr)
+		VISITOR_STMT(UnresolvedMemberExpr)
+		VISITOR_STMT(UserDefinedLiteral)
+		VISITOR_STMT(VAArgExpr)
+		VISITOR_STMT(WhileStmt)
+	};
+
+	template< typename Derived >
+	struct UnsupportedTypeVisitor
+		: clang::TypeVisitor< UnsupportedTypeVisitor< Derived >, vast::hl::Type >
+		, vast::cg::CodeGenVisitorLens< UnsupportedTypeVisitor< Derived >, Derived >
+		, vast::cg::CodeGenBuilderMixin< UnsupportedTypeVisitor< Derived >, Derived >
+	{
+
+		using LensType = vast::cg::CodeGenVisitorLens< UnsupportedTypeVisitor< Derived >, Derived >;
+		using base_type = clang::TypeVisitor< UnsupportedTypeVisitor< Derived >, Derived >;
+
+		using LensType::derived;
+		using LensType::context;
+		using LensType::mcontext;
+		using LensType::acontext;
+		using LensType::visit;
+
+		vast::hl::Type VisitTypeOfType(const clang::TypeOfType *type) {
+			return visit(type->desugar());
+		}
+
+		vast::hl::Type VisitMacroQualifiedType(const clang::MacroQualifiedType *type) {
+			return visit(type->desugar());
+		}
+	};
+
+	template< typename Derived >
+	struct UnsupportedFallBackVisitor
+	   	: UnsupportedDeclVisitor< Derived >
+	   	, UnsupportedStmtVisitor< Derived >
+	   	, UnsupportedTypeVisitor< Derived >
+	{
+	   	using DeclVisitor = UnsupportedDeclVisitor< Derived >;
+	    using StmtVisitor = UnsupportedStmtVisitor< Derived >;
+	    using TypeVisitor = UnsupportedTypeVisitor< Derived >;
+
+	    using DeclVisitor::Visit;
+	    using StmtVisitor::Visit;
+	    using TypeVisitor::Visit;
+
+	    operation Visit(const clang::Stmt *stmt) {
```

Comment:
Why are these methods here? Except for the `Type` overload, which has custom logic.

Reply:
Removed this additional indirection!